main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlService.cs

```
@@ -847,6 +846,12 @@ public RepositoryMetadata ()
 		{
 		}
 
+		public RepositoryMetadata (VersionControlSystem versionControl)
```

We should explicitly disallow the normal constructor, as we could get events that get the metadata bag created via the Counter itself, thus circumventing this.

---

My point:
```
using (var counter = CreateCounter<RepositoryMetadata>())
{
   counter.Metadata.Path = value;
}
```

Will send over a counter with no VCS system set. We should prevent not having one set.

---

That's not easy because of https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Core/MonoDevelop.Core.Instrumentation/InstrumentationService.cs#L247, but the metadata should not be public and we could throw an `InvalidOperationException` internally.

---

Yup! throw InvalidOperationException in the normal constructor

---

done in 29e31b5